libcudacxx/test/libcudacxx/cuda/memory_resource/device_memory_resource/equality.pass.cpp

```diff
@@ -30,7 +30,7 @@ struct resource
   {
     return nullptr;
   }
-  void deallocate(void*, size_t, size_t) {}
+  void deallocate(void*, size_t, size_t) noexcept {}
```

Comment:
I am not a fan of this churn, why is this necessary?

Reply:
i thought it would be a good idea for `resource_ref` to `static_assert` that `deallocate` functions were `noexcept`. i still think it's a good idea, but i had to walk back that change because RMM's allocators don't mark their `deallocate` members as `noexcept`.

my plan is to submit a pr to RMM to make it conform and then re-enable the check, but i could go the other way and back all of these changes out. thoughts?